contracts/ChildChainGaugeInjectoooor.sol

```diff
@@ -85,21 +78,20 @@ contract ChildChainGaugeInjector is ConfirmedOwner, Pausable, KeeperCompatibleIn
         uint8[] calldata maxPeriods
     ) public onlyOwner {
         if (gaugeAddresses.length != amountsPerPeriod.length || gaugeAddresses.length != maxPeriods.length) {
-            revert InvalidGaugeList();
+            revert InvalidGaugeList("supplied gauges are different lengths");
         }
         address[] memory oldGaugeList = s_gaugeList;
         for (uint256 idx = 0; idx < oldGaugeList.length; idx++) {
             s_targets[oldGaugeList[idx]].isActive = false;
         }
+        revertOnDuplicate(gaugeAddresses);
```

Comment:
L94: The duplicate check in setRecipientList inspects a storage var (s_targets) inside a loop. It is much more efficient to do this by comparing memory var (gaugeAddresses) elements inside a nested loop.